lib/scope.js

```diff
@@ -450,13 +440,25 @@ AST_Toplevel.DEFMETHOD("mangle_names", function(options){
             to_mangle.push(node.definition());
             return;
         }
-    });
-    this.walk(tw);
+    }));
     to_mangle.forEach(function(def){ def.mangle(options) });
 
     if (options.cache) {
         options.cache.cname = this.cname;
     }
+
+    function push_symbols(dict) {
+        var a = [];
+        dict.each(function(symbol) {
+            if (options.reserved.indexOf(symbol.name) < 0) {
+                a.push(symbol);
+            }
+        });
+        a.sort(function(m, n) {
+            return n.references.length - m.references.length;
```

Comment:
`figure_out_scope()` is called again after `compress()` and right before `mangle_names()`, as I mentioned in the code references in https://github.com/mishoo/UglifyJS2/pull/2219#issuecomment-313864600

... or have I missed something? :sweat_smile: